### PR TITLE
Add glossary entries for needed terms.

### PR DESCRIPTION
**Pull request by rcaballeromx**
The terms involved are:

    - Cluster
    - Identity
    - Trust domain

Signed-off-by: rcaballeromx <grca@google.com>

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ X ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/reference/glossary/identity.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+title: Identity
+---
+
+An identity is a unique name backed by a key material, for
```
Comment (geeknoid) [inline content/en/docs/reference/glossary/identity.md]:
```suggestion
An identity is a unique name backed by key material, for
```

File: content/en/docs/reference/glossary/cluster.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+title: Cluster
```
Comment (nmittler) [inline content/en/docs/reference/glossary/cluster.md]: I'm assuming you borrowed liberally from Kubernetes here. I don't think we should use their definition. We need to be more generic ... Istio is not k8s-specific, and neither is the concept of a cluster.

File: content/en/docs/reference/glossary/identity.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+example:
+
+- [X.509 certificate](https://en.wikipedia.org/wiki/X.509)
+- [JWT](https://en.wikipedia.org/wiki/JSON_Web_Token)
+
+Each identity Istio assigns is understood throughout the mesh and you can use it
+to enable mutual authentication and enforce policies.
+
+An identity contains a substring identifying the mesh that created it. For
```
Comment (nmittler) [inline content/en/docs/reference/glossary/identity.md]: You're duplicating the definition of trust domain here.  We should probably remove.

File: content/en/docs/reference/glossary/trust-domain.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+title: Trust Domain
+---
+
+An identity contains a substring identifying the mesh that created it. For
```
Comment (nmittler) [inline content/en/docs/reference/glossary/trust-domain.md]: should you have identity be a link?

File: content/en/docs/reference/glossary/identity.md (hunk 3 above)
Comment (frankbu) [reply content/en/docs/reference/glossary/identity.md]: Or replace it with a link to the Trust Domain entry.

File: content/en/docs/reference/glossary/cluster.md (hunk 2 above)
Comment (frankbu) [reply content/en/docs/reference/glossary/cluster.md]: I agree, this needs to be a more generic definition of cluster, not the K8s one.

Comment (rcaballeromx) [reply content/en/docs/reference/glossary/cluster.md]: I'm replacing the definitions with the following platform agnostic version: 
```
A set of compute nodes within physical proximity that run containerized
applications.

Typically, the compute nodes comprising a cluster can reach each other directly.
```
WDYT?

Comment (nmittler) [reply content/en/docs/reference/glossary/cluster.md]: I think that's a lot better ... let's switch to this and let folks throw darts at it.

File: content/en/docs/reference/glossary/trust-domain.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+---
+title: Trust Domain
+---
+
+An [identity](/docs/reference/glossary/#identity) contains a substring
```
Comment (nmittler) [inline content/en/docs/reference/glossary/trust-domain.md]: We should probably lead with the definition of trust domain.  Maybe something like:

```
A trust domain is a globally unique name used to create identities. Every Istio mesh has exclusive ownership of a single trust domain, which is used to create all identities within the mesh. 
```

I think that should be sufficient.  WDYT?

Comment (frankbu) [reply content/en/docs/reference/glossary/trust-domain.md]: @nmittler Yes, I like your wording but would still add a link:
```
A trust domain is a globally unique name used to create
[identities](/docs/reference/glossary/#identity). Every Istio
mesh has exclusive ownership of a single trust domain, which is used to create
all identities within the mesh.
```

Comment (rcaballeromx) [reply content/en/docs/reference/glossary/trust-domain.md]: Maybe a simplified version:

```
A trust domain is a globally unique name that Istio uses to create all identities within a mesh. Every mesh has an exclusive trust domain. 
```
WDYT

Comment (rcaballeromx) [reply content/en/docs/reference/glossary/trust-domain.md]: And yes definitely with a link.

Comment (nmittler) [reply content/en/docs/reference/glossary/trust-domain.md]: Hmm ... yours is more brief, for sure. But I feel it loses the explicit statement that all identities in the mesh are created with the same trust domain. Yes it's implied, but you have to parse it in your head a little to get there.

Comment (nmittler) [reply content/en/docs/reference/glossary/trust-domain.md]: Actually, looking at it again, I think your definition is fine. 

@frankbu WDYT?

Comment (frankbu) [reply content/en/docs/reference/glossary/trust-domain.md]: Yep, the simpler version looks fine to me as long as it has the link from "identities".

**Comment by nmittler** [comment]
@smawson @myidpt PTAL

**Comment by rcaballeromx** [comment]
I'm closing this PR to and including the entry glossaries in #4813

**Pull request closed without merging**